e2e/conftest.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import annotations
+
+import datetime
+import logging
+import random
+import socket
+import subprocess
+import sys
+import threading
+from pathlib import PurePath
+from time import sleep
+from types import TracebackType
+from typing import IO, Callable, Generator, List, Optional, TextIO, Type, Union
+
+import pytest
+
+__all__ = (
+    "ShinyAppProc",
+    "create_app_fixture",
+    "create_example_fixture",
+    "local_app",
+    "run_shiny_app",
+)
+
+here = PurePath(__file__).parent
+
+
+def random_port():
+    while True:
+        port = random.randint(1024, 49151)
+        with socket.socket() as s:
+            try:
+                s.bind(("127.0.0.1", port))
+                return port
+            except Exception:
+                # Let's just assume that port was in use; try again
+                continue
```

Comment:
Could merge #329 first and get rid of this?

Reply:
Dang it, I _knew_ I had written this somewhere already! 😆

Reply:
I'll unify them afterwards just so we can get this in now.